File: SpotifyTrendsApp.Server/Models/Track.cs
using System;

namespace SpotifyTrendsWebApp.Server.Models;

public class TrackResponse
{
    public string href { get; set; }
    public int limit { get; set; }
    public List<Track> items { get; set; }
    public int offset { get; set; }
    public int total { get; set; }
}

public class Track
    {
        public Album album { get; set; }
        public IEnumerable<Artist> artists { get; set; }
        public IEnumerable<string> available_markets { get; set; }
        public int disc_number { get; set; }
        public int duration_ms { get; set; }
        public bool _explicit { get; set; }
        public string href { get; set; }
        public string id { get; set; }
        public bool is_playable { get; set; }
        public string name { get; set; }
        public int popularity { get; set; }
        public object preview_url { get; set; }
        public int track_number { get; set; }
        public string type { get; set; }
        public string uri { get; set; }
        public bool is_local { get; set; }
    }




 

    

File: go-services/pkg/models/track.go
package models

// Track represents a Spotify track
type Track struct {
	ID               string   `json:"id"`
	Name             string   `json:"name"`
	Album            Album    `json:"album,omitempty"`
	Artists          []Artist `json:"artists,omitempty"`
	AvailableMarkets []string `json:"available_markets,omitempty"`
	DiscNumber       int      `json:"disc_number,omitempty"`
	DurationMs       int      `json:"duration_ms,omitempty"`
	Explicit         bool     `json:"explicit,omitempty"`
	Href             string   `json:"href,omitempty"`
	IsPlayable       bool     `json:"is_playable,omitempty"`
	Popularity       int      `json:"popularity,omitempty"`
	PreviewURL       *string  `json:"preview_url,omitempty"`
	TrackNumber      int      `json:"track_number,omitempty"`
	Type             string   `json:"type,omitempty"`
	URI              string   `json:"uri,omitempty"`
	IsLocal          bool     `json:"is_local,omitempty"`
}

// TrackResponse represents the paginated track response from Spotify
type TrackResponse struct {
	Href   string  `json:"href"`
	Limit  int     `json:"limit"`
	Items  []Track `json:"items"`
	Offset int     `json:"offset"`
	Total  int     `json:"total"`
	Next   *string `json:"next"`
	Prev   *string `json:"previous"`
}
